config/initializers/redis.rb

```diff
@@ -0,0 +1 @@
+$redis = Redis::Namespace.new('todo_list', redis: Redis.new)
```

Comment:
Considering the fact that you already have Rails cache store wired to redis, I think you don't need to this raw redis connection. You can just
```ruby
Rails.cache.fetch('todos') { Todo.all }
Rails.cache.fetch("todos/#{id}") { Todo.find(id) }
```